inference-engine/src/mkldnn_plugin/nodes/mkldnn_rnn.cpp

```diff
@@ -248,119 +278,149 @@ void MKLDNNRNN::fillSeqDesc() {
     if (out_data_dims != OD_shape)
         IE_THROW() << "Incorrect shape of input/output ports for layer " << getName();
 
-    in_states_d.resize(S);
-    out_states_d.resize(S);
+    auto& blobs = rnnLayer->blobs;
+    Blob::Ptr weights, bias;
+    if (blobs.find("weights") != blobs.end()) weights = blobs["weights"];
+    if (blobs.find("biases") != blobs.end()) bias = blobs["biases"];
+
+    if (!weights)
+        IE_THROW() << "RNN Layer. Weights do not present.";
+
+    if (weights->size() != G * SC * (SC + DC))
+        IE_THROW() << "RNN Layer. Weights size is not correct. Expected size:" << G * SC * (SC + DC);
 
     for (int i = 1; i < ins.size(); i++) {
         if (getParentEdgeAt(i)->getDims() != S_shape)
             IE_THROW() << "Incorrect shape of state ports for layer " << getName();
-        in_states_d[i - 1] = {S_4D_shape, memory::data_type::f32, memory::format_tag::ldnc};
     }
 
     for (int i = 1; i < outs.size(); i++) {
         if (getChildEdgeAt(i)->getDims() != S_shape)
             IE_THROW() << "Incorrect shape of state ports for layer " << getName();
-        out_states_d[i - 1] = {S_4D_shape, memory::data_type::f32, memory::format_tag::ldnc};
     }
 
-    auto blobs = rnnLayer->blobs;
-    Blob::Ptr weights, bias;
-    if (blobs.find("weights") != blobs.end()) weights = blobs["weights"];
-    if (blobs.find("biases") != blobs.end()) bias = blobs["biases"];
+    // layer input plus states
+    in_data_d.resize(S + 1);
+    out_data_d.resize(S + 1);
 
-    if (!weights)
-        IE_THROW() << "RNN Layer. Weights do not present.";
+    auto runtimePrecision = outs[0]->getPrecision();
```

Comment:
You still get runtime precision from output here. Does it make sense to:
1. Make runtime precision as node field?
2. Move its initialization and verification into getSupportedDescriptors since it seems to be equal for both fillers?